### DISCUSSION TITLE
Numbered and cross-referencable Remarks

### DISCUSSION THREAD

**Discussion by AlbertRapp** [quarto-dev/quarto-cli#3322] [Feature Requests]
Would it be possible to transform the `.remark` class into a `#rem-` name such that remarks are numbered and become cross-referenceable?

According to the [docs](https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs) this is intended behavior.
In mathematics, we use remarks all the time and cross-reference them too. That's easy in LaTeX. Would be great if that's possible in Quarto too.

**Comment by cscheid** [comment]
We hope to do work on crossrefs in 1.3, so I went ahead and created an issue to track: #3323

**Reply by AlbertRapp** [reply]
That's awesome. Thank you. Happy to hear that. ☺️

**Comment by AlbertRapp** [comment]
Just in case someone is stumbling on this before remarks become cross-referencable by default. Here's my current workaround.
It relies on using the conjecture environment (which I do not need) and renaming it after use.

1. Use `#cnj-` environment whenever you want to create a Remark.
2. Add a custom class to your div. In my case this would look like `::: {#cnj-my-remark .custom}`.
4. Create a `rename_script.html` file that contains the following code
    ```
    <script>
    var custom_title = document.querySelectorAll('.custom .theorem-title');
    for (let i = 0; i < custom_title.length; i++ ) {
       var mod_name = custom_title[i].innerHTML;
       custom_title[i].innerHTML = mod_name.replace("Conjecture", "Remark");
    };

    var cnj_citations = document.querySelectorAll("a[href*='#cnj']");
    for (let i = 0; i < cnj_citations.length; i++ ) {
       var cite_name = cnj_citations[i].innerHTML;
       cnj_citations[i].innerHTML = cite_name.replace("Conjecture", "Remark");
    };
    </script>
    ```
4. Include the script in your Quarto doc in the header with
    ```
    format:
        html:
            include-after-body: rename_script.html
    ```
5. The first part of your script will target the `<span>`s (inline texts) that have a `.custom` and `.theorem-title` class tag and replace the word "Conjecture" by "Remark". So if you're writing in a different language you will have to change the script. For example, in German I use `mod_name.replace("Annahme", "Bemerkung");`. The second part of the script will target `<a>`s (links) that contain `#cnj` in the URL they link to. These are the links from your cross-references. We have to replace the word "Conjecture" by "Remark" there too.

Hope this helps someone struggling with the same thing I am.
And a warning: I do not really know any Javascript, so my script chunk may be rubbish. I've build this with code snippets I found online.

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/3323
